### PR TITLE
Implement `array_pop_back` function 

### LINKED ISSUE #6997: Implement `array_pop_back` function
### Is your feature request related to a problem or challenge?

## Summary

| Characteristic        | Description                                                                                                                                                                                                                     |
| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Function name:        | `array_pop_back`                                                                                                                                                                                                                |
| Aliases:              | `list_pop_back`                                                                                                                                                                                                                 |
| Original function?:   | No                                                                                                                                                                                                                              |
| Function Description: | <b>ClickHouse:</b> Removes the last item from the array. </br> <b>DuckDB:</b> Returns the list without the last element.                                                                                                        |
| Sources:              | [Concept](https://github.com/apache/arrow-datafusion/discussions/6855) [DuckDB](https://duckdb.org/docs/sql/functions/nested) [ClickHouse](https://clickhouse.com/docs/en/sql-reference/functions/array-functions#arraypopback) |

## Examples:

```
D select array_pop_back([1, 2, 3, 4]);
┌─────────────────────────────────────────────┐
│ array_pop_back(main.list_value(1, 2, 3, 4)) │
│                   int32[]                   │
├─────────────────────────────────────────────┤
│ [1, 2, 3]                                   │
└─────────────────────────────────────────────┘
D select array_pop_back([[1, 2], [3, 4], [4, 2]]);
┌──────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ array_pop_back(main.list_value(main.list_value(1, 2), main.list_value(3, 4), main.list_value(4, 2))) │
│                                              int32[][]                                               │
├──────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ [[1, 2], [3, 4]]                                                                                     │
└──────────────────────────────────────────────────────────────────────────────────────────────────────┘
```


### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tanruixiang**
## Which issue does this PR close?

Closes #6997 .

## Rationale for this change
see #6997 

## Are these changes tested?
yes

### CODE REVIEW

**Comment by tanruixiang** [comment]
cc @izveigor  @jayzhan211

File: datafusion/physical-expr/src/array_expressions.rs [hunk 1]
```diff
@@ -601,6 +601,22 @@ pub fn array_slice(args: &[ArrayRef]) -> Result<ArrayRef> {
     define_array_slice(list_array, key, extra_key, false)
 }
 
+pub fn array_pop_back(args: &[ArrayRef]) -> Result<ArrayRef> {
```
[the remark this answers is not in the document]
Comment (tanruixiang) [reply datafusion/physical-expr/src/array_expressions.rs:604]: Yes, I would love to implement `array_pop_front `, do I need to implement `array_pop_front ` in this pr? Or in the next pr.

[the remark this answers is not in the document]
Comment (jayzhan211) [reply datafusion/physical-expr/src/array_expressions.rs:604]: Next pr.

**Review by alamb — APPROVED** [review]
LGTM too -- thank you @tanruixiang 

I think this PR just needs to have conflicts resolved and it should be good to merge

Thank you @jayzhan211 and @izveigor  for reviews

**Comment by tanruixiang** [comment]
@alamb @jayzhan211 @izveigor Thanks to all for the reviews, the conflict has been resolved!

**Comment by alamb** [comment]
Thanks @tanruixiang

**Pull request merged**